models/detr.py

```diff
@@ -161,6 +162,22 @@ def loss_boxes(self, outputs, targets, indices, num_boxes):
         losses['loss_giou'] = loss_giou.sum() / num_boxes
         return losses
 
+    def convert_polygon2mask(self, targets, size, indices):
+        masks = []
+        b, m = indices
+        for i in range(len(b)):
+            t = targets[b[i]]
+            h, w = t["size"]
+            polygon = t["masks"].polygons[m[i]]
+            for p in polygon:
+                p[0::2] *= (size[1] / w)
+                p[1::2] *= (size[0] / h)
+            masks.append(polygons_to_bitmask(polygon, size[0], size[1]))
+        masks = torch.tensor(masks)
+        if len(masks.shape) != 3:
+            masks = masks.unsqueeze(0)
+        return masks
```

Comment:
This should not live in `models/detr`, but instead in the d2 wrapper code to convert to the representation that we expect.

This way, the model in `detr` only expects a single type of input data, making things simpler overall.

Reply:
Then, maybe making inheritance `SetCriterion` on `d2/detr` folder from `SetCriterion` on `models/detr.py` will be better.

Reply:
I believe this should be changed in `prepare_targets` in https://github.com/facebookresearch/detr/blob/f4cdc542de34de771da8b9189742e5465f5220cd/d2/detr/detr.py#L160

You can add one extra line in the implementation in https://github.com/facebookresearch/detr/blob/f4cdc542de34de771da8b9189742e5465f5220cd/d2/detr/detr.py#L179-L188 to convert the polygons to bitmask